studio/components/interfaces/Organization/Usage/Usage.tsx

```diff
@@ -1,29 +1,44 @@
 import clsx from 'clsx'
 import dayjs from 'dayjs'
-import { useMemo, useState } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 
 import { useParams } from 'common'
-import { ScaffoldContainer, ScaffoldDivider } from 'components/layouts/Scaffold'
+import { ScaffoldContainer } from 'components/layouts/Scaffold'
 import DateRangePicker from 'components/to-be-cleaned/DateRangePicker'
 import AlertError from 'components/ui/AlertError'
 import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import { useProjectsQuery } from 'data/projects/projects-query'
 import { useOrgSubscriptionQuery } from 'data/subscriptions/org-subscription-query'
+import { useSelectedOrganization } from 'hooks'
 import { TIME_PERIODS_BILLING, TIME_PERIODS_REPORTS } from 'lib/constants'
+import { Listbox } from 'ui'
 import Activity from './Activity'
 import Bandwidth from './Bandwidth'
 import SizeAndCounts from './SizeAndCounts'
 
 const Usage = () => {
-  const { slug } = useParams()
+  const { slug, projectRef } = useParams()
   const [dateRange, setDateRange] = useState<any>()
+  const [selectedProjectRef, setSelectedProjectRef] = useState<string>()
 
+  const organization = useSelectedOrganization()
+  const { data: projects, isSuccess } = useProjectsQuery()
   const {
     data: subscription,
     error: subscriptionError,
     isLoading: isLoadingSubscription,
     isError: isErrorSubscription,
     isSuccess: isSuccessSubscription,
   } = useOrgSubscriptionQuery({ orgSlug: slug })
+  const orgProjects = projects?.filter((project) => project.organization_id === organization?.id)
+
+  useEffect(() => {
+    if (projectRef && isSuccess && orgProjects !== undefined) {
+      if (orgProjects.find((project) => project.ref === projectRef)) {
```

Comment:
Is there a risk of `orgProjects` becoming stale here? On solution might be to `const projectsRef = useLatest(orgProjects)`

Reply:
hmm i dont think so - is there a reason why you might thinking that `orgProjects` might become stale though? in case im overlooking anything here

i mainly dont think so because this is just to check if the URL does have ?projectRef=xxx provided (used for linking from the project's general settings page). the param isn't expected to change while on the org settings page